src/main/java/com/efficio/fieldbook/web/common/controller/CrossingSettingsController.java

```diff
@@ -263,6 +266,49 @@ public String getContentName() {
 
 	}
 
+	/**
+	 * Validates the Breeding Methods in the import file
+	 *
+	 * @return a JSON result object
+	 */
+	@ResponseBody
+	@RequestMapping(value = "/validateBreedingMethods/{isBasedOnImportFile}/{breedingMethodId}", method = RequestMethod.GET)
+	public Map<String, Object> validateBreedingMethods(@PathVariable final boolean isBasedOnImportFile, @PathVariable final int breedingMethodId) {
+		final Map<String, Object> out = new HashMap<>();
+
+		if(isBasedOnImportFile) {
+			final Set<String> breedingMethods = this.studySelection.getImportedCrossesList().getImportedCrosses().stream()
+				.filter(cross -> !StringUtils.isEmpty(cross.getRawBreedingMethod())).map(ImportedCross::getRawBreedingMethod).collect(
+					Collectors.toSet());
+			if(!CollectionUtils.isEmpty(breedingMethods)) {
+				final List<String> nonGenerativeBreedingMethodCodes =
+					this.germplasmDataManager.getNonGenerativeMethodCodes(breedingMethods);
```

Comment:
Instead of creating new methods in DAO (getNonGenerativeMethodCodes and methodCodesWithOneMPGRN) that are specific for validation, you can utilize org.generationcp.middleware.api.breedingmethod.BreedingMethodServiceImpl#getBreedingMethods to get the list of methods based on method types. And use that list to validate here in the controller, something like: 

```
// Get Derivative and Maintenance Methods
				final List<BreedingMethodDTO> breedingMethodDTOS = this.breedingMethodService
					.getBreedingMethods(new BreedingMethodSearchRequest(null,
						Arrays.asList(MethodType.DERIVATIVE.getCode(), MethodType.MAINTENANCE.getCode()), false));

				final Set<String> derivativeMethodsMatches =
					breedingMethodDTOS.stream().filter(dto -> breedingMethods.contains(dto.getCode())).map(BreedingMethodDTO::getCode)
						.collect(Collectors.toSet());
				final Set<String> derivativeMethodsWithOnlyOneProgenitorMatches = breedingMethodDTOS.stream()
					.filter(dto -> breedingMethods.contains(dto.getCode()) && dto.getNumberOfProgenitors().intValue() == 1).
						map(BreedingMethodDTO::getCode).collect(Collectors.toSet());

				if (!breedingMethods.equals(derivativeMethodsMatches)) {
					out.put(CrossingSettingsController.ERROR, this.messageSource.getMessage("error.crossing.non.generative.method",
						new String[] {
							StringUtils.join(breedingMethods.stream().filter(val -> !derivativeMethodsMatches.contains(val))
								.collect(Collectors.toList()), ", ")}, LocaleContextHolder.getLocale()));
				} else if (!CollectionUtils.isEmpty(derivativeMethodsWithOnlyOneProgenitorMatches)) {
					out.put(CrossingSettingsController.ERROR, this.messageSource.getMessage("error.crossing.method.mprgn.equals.one",
						new String[] {StringUtils.join(derivativeMethodsWithOnlyOneProgenitorMatches, ", ")},
						LocaleContextHolder.getLocale()));
				}
```

Let me know if this works.